gomanifest/build.go

```diff
@@ -0,0 +1,234 @@
+}
+
+// GoPackage ... Package structure from deps json
+type GoPackage struct {
+	Root       string   `json:"Root"`
+	ImportPath string   `json:"ImportPath"`
+	Module     GoModule `json:"Module"`
+	Standard   bool     `json:"Standard"`
+	Imports    []string `json:"Imports"`
+	Deps       []string `json:"Deps"`
+}
+
+/* Structure for output the manifest */
+
+// ManifestTransitiveDeps ... Transitive details in final manifest
+type ManifestTransitiveDeps struct {
+	Name    string `json:"package"`
+	Version string `json:"version"`
+}
+
+// MainfestDirectDeps ... Direct dependency details.
+type MainfestDirectDeps struct {
+	Name         string                   `json:"package"`
+	Version      string                   `json:"version"`
+	Dependencies []ManifestTransitiveDeps `json:"deps"`
+}
+
+// Manifest ... Final manifest file structure
+type Manifest struct {
+	Version  string               `json:"version"`
+	Main     string               `json:"main"`
+	Packages []MainfestDirectDeps `json:"packages"`
+}
+
+// Deps package from `go list -deps -json ./...` command
+var depsPackages = make(map[string]GoPackage)
+
+// transformVersion ... Converts the golang version string into semver without 'v' and appended text after '+'
+func transformVersion(inVersion string) string {
+	var outVersion string = strings.Replace(inVersion, "v", "", 1)
+
+	return strings.Split(outVersion, "+")[0]
+}
+
+// getPackageName ... Utility function to convert package + module data into package name used by manifest.
+func getPackageName(depPackage GoPackage) string {
+	// Get module / package@module
+	if depPackage.ImportPath != depPackage.Module.Path {
+		return depPackage.ImportPath + "@" + depPackage.Module.Path
+	}
+
+	return depPackage.ImportPath
+}
+
+// GoListCmd ... Go list command structure.
+type GoListCmd struct {
+	CWD string
+}
+
+// Run ... Actual function that executes go list command and returns output as string.
+func (goListCmd GoListCmd) Run() (string, error) {
+	GoListGoListDeps := exec.Command("go", "list", "-json", "-deps", "./...")
+	GoListGoListDeps.Dir = goListCmd.CWD
+	output, err := GoListGoListDeps.Output()
+
+	if err != nil {
+		return "", err
+	}
+
+	fmt.Println("Outp", string(output))
+	return string(output), nil
+}
+
+// GoListCmdInterface ... Interface to be implemented to execute go list command.
+type GoListCmdInterface interface {
+	Run() (string, error)
+}
+
+// GoList ... Structure that handle go list data and extract required packages.
+type GoList struct {
+	Command GoListCmdInterface
+}
+
+// Get ... Get deps data through go list deps command and converts json into objects.
+func (goList *GoList) Get() int {
+	output, err := goList.Command.Run()
+
+	if err != nil {
+		log.Println("ERROR :: Command `go list -json -deps ./...` failed, resolve project build errors.", err)
+		return -1
+	}
+
+	goListDepsData := string(output)
+	goListDepsData = `{"Packages": [` + strings.ReplaceAll(goListDepsData, "}\n{", "},\n{") + "]}"
```

Comment:
I have checked package you shared on slack but that was a code that rely on import from the source similar to what we do in `go list -deps -json` command. We have lot of use-case to extract deps data for go project and I still feel go list is good command than using any other custom code.

Reply:
I didn't suggest any replacement for `go list -deps -json`, I asked to use streaming json parser to avoid patching up for missing comma for each package entry,

Reply:
This is done.